internal/graph/object_providers.go

```diff
@@ -79,9 +75,6 @@ func (c *recursiveTTUObjectProvider) End() {
 	if c.cancel != nil {
 		c.cancel()
 	}
-	if c.pool != nil {
```

Comment:
there is a change in behaviour in that we don't wait for the goroutine to be completed.  Is that intentional?

Reply:
Correct. This made any short circuiting have to wait for cleanup, def not something we want.